### DISCUSSION TITLE
How update existing environment's OpenAPI Specification file?

### DISCUSSION THREAD

**Discussion by kaushil24** [mockoon/mockoon#657] [Help] [answered]
Some of our endpoints are constantly evolving and as a result, our OpenAPI Specification file gets updated frequently. Let's say, I have already created an environment by importing an OpenAPI Spec file and added a bunch of rules, templates, and response bodies. Now, say my OpenAPI Spec file has changed. How can I re-import the updated spec file into my existing mockoon environment such that I don't loose the response templates, rules etc? 

I did not find any documentation to handle such a scenario. Does mockoon provide such a mechanism? If not, then is there a workaround for that?

Thanks!

**Comment by 255kb** [comment, marked as answer]
There is no such mechanism as "Open API spec update". 
Mockoon was built from the beginning as a quick way to mock REST APIs but not as an API design tool. 
So, the OpenAPI specs import is currently quite basic. There is no feature allowing for a reimport of the specification file and merge with an existing Mockoon's environment. 

The only "workaround" is to reimport the specification in a new environment and copy the missing routes. But for the existing routes there is no merge option. 
Also, this is not in the roadmap and probably wouldn't be a priority considering other features that are way more requested and sometimes already under development (like GraphQL mocks).
